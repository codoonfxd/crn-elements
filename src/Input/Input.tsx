/*
 * @Author: liuyz
 * @Date: 2019-04-17 17:28:47
 * @Last Modified by: liuyz
 * @Last Modified time: 2019-04-18 11:44:01
 */

import * as React from 'react'
import { TextInput, ViewStyle, TextInputProperties } from 'react-native'
import Color from 'color'
import { THEME_COLOR } from '../lib/constant'

interface IProps extends TextInputProperties {
  style?: ViewStyle
  onChangeText(text: string): any
}

export interface IState {
  text: string
}

export default class Input extends React.Component<IProps, IState> {
  inputRef: React.ReactNode
  constructor(props: IProps) {
    super(props)

    this.state = {
      text: '',
    }
  }

  textChange(text: string): void {
    this.setState({ text })
    if (this.props.onChangeText) {
      this.props.onChangeText(text)
    }
  }

  render() {
    const {
      placeholder = '请输入内容',
      selectionColor,
      style = {},
    } = this.props
    const color = Color(style.borderColor || THEME_COLOR)
    const borderColor = color.string()

    const { text } = this.state
    return (
      <TextInput
        ref={(el: React.ReactNode) => (this.inputRef = el)}
        placeholder={placeholder}
        selectionColor={selectionColor || THEME_COLOR} // 设置输入框高亮时的颜色（在iOS上还包括光标）
        underlineColorAndroid="transparent"
        // androidnumberOfLines={3} // 设置输入框的行数 当multiline设置为true时使用它
        {...this.props}
        style={{
          height: 36,
          paddingHorizontal: 8,
          borderColor,
          borderWidth: 1,
          borderRadius: 4,
          ...style,
        }}
        onChangeText={(value) => this.textChange(value)}
        value={text}
      />
    )
  }
}
