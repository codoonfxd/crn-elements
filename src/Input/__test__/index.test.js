import * as React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

describe('<Input />', () => {
  it('渲染快照', () => {
    const tree = renderer.create(<Input placeholder="输入框" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
