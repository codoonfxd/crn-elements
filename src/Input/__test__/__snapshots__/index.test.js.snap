// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> 渲染快照 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder="输入框"
  rejectResponderTermination={true}
  selectionColor="#00bc71"
  style={
    Object {
      "borderColor": "rgb(0, 188, 113)",
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 36,
      "paddingHorizontal": 8,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
